src/main/java/Duke.java

```diff
@@ -1,10 +1,125 @@
+import java.util.Scanner;
 public class Duke {
+    private static final int LIST_SIZE = 100;
+    private static int taskNum = 0;
     public static void main(String[] args) {
+        showGreetings();
+        String command = "";
+        Scanner in = new Scanner(System.in);
+
+        Task[] storedTasks = new Task[LIST_SIZE];         // array of stored texts
+        while (!command.equals("bye")) {
+            printLineBreak();
+            command = in.nextLine();
+            String[] commandLine = command.split(" ", 2);
+
+            printLineBreak();
+
+            switch (commandLine[0]) {
+            case "todo":
+                addTodoTask(storedTasks, commandLine);
+                break;
+            case "deadline":
+                addDeadlineTask(storedTasks, commandLine);
+                break;
+            case "event":
+                addEventTask(storedTasks, commandLine);
+                break;
+            case "mark":
+                markTask(storedTasks, commandLine);
+                break;
+            case "unmark":
+                unmarkTask(storedTasks, commandLine);
+                break;
+            case "list":
+                listTasks(storedTasks);
+                break;
+            }
+
+        }
+
+        System.out.println("Bye. Hope to see you again soon!");
+
+    }
+
+    private static void listTasks(Task[] storedTasks) {
+        listMessage();
+        for (int i = 0; i < taskNum; i++) {
+            System.out.println((i + 1) + ". " + storedTasks[i].getTypeIcon() +
+                    storedTasks[i].getStatusIcon() + " " + storedTasks[i].getDescription());
+        }
+    }
+
+    private static void unmarkTask(Task[] storedTasks, String[] commandLine) {
+        int unmarkIndex = Integer.parseInt(commandLine[1]) - 1;
+        storedTasks[unmarkIndex].setDone(false);
+        System.out.println("OK, I've marked this task as not done yet.");
+        System.out.println(storedTasks[unmarkIndex].getTypeIcon() +
+                storedTasks[unmarkIndex].getStatusIcon() + " " + storedTasks[unmarkIndex].getDescription());
+    }
+
+    private static void markTask(Task[] storedTasks, String[] commandLine) {
+        int taskIndex = Integer.parseInt(commandLine[1]) - 1;
+        storedTasks[taskIndex].setDone(true);
+        System.out.println("Nice! I've marked this task as done.");
+        System.out.println(storedTasks[taskIndex].getTypeIcon() +
+                storedTasks[taskIndex].getStatusIcon() + " " + storedTasks[taskIndex].getDescription());
+    }
+
+    private static void addEventTask(Task[] storedTasks, String[] commandLine) {
```

Comment:
Great work with refactoring out multiple functions that is able to add tasks!

Reply:
Thanks